source/language/openpulse.rst

```diff
@@ -30,227 +31,234 @@ could even reference multiple qubits in a chain.
-names provided a mapping to the appropriate channels is given.
+- Tracking time appropriately so programs do not need to deal in absolute time
+- Tracking phase by producing a complex value given an input time (i.e. via the mathematical
+  relationship :math:`e^{i\left(2\pi f t + \theta\right)}`,  where `f` is frequency and
+  :math:`\theta` is phase). One motivation for keeping track of phase is to allow pulses to be be
+  defined in the rotating frame (i.e. without the carrier) with the effect being
+  an equivalent application in the lab frame (i.e. with the carrier supplied by the ``frame``).
+  Another motivation is to more naturally implement a "virtual Z-gate", which does not require a
+  physical pulse but rather shifts the phase of all future pulses on that frame.
 
-Restrictions on the use of frames
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The frame is composed of three parts:
 
-There are two important considersations when dealing with frames.
+1. A frequency ``frequency`` of type ``float``.
+2. A phase ``phase`` of type ``angle``.
+3. A time of type ``dt`` which is manipulated implicitly and cannot be modified other
+   than through existing timing instructions like ``delay`` and ``barrier``.
 
-First, frames are compile time constants. Runtime variables cannot be used as
-arguments to frames; the frame must be resolvable at compile time. This also
-means that assigning a frame to an alias (for the purpose of avoiding typing)
-needs to be done with ``const`` not ``let``.
+The ``frame`` type is a virtual resource and the exact precision of these parameters is
+hardware specific. It is thus up to the compiler to choose how to implement the required
+transformations to physical resources in hardware (e.g. mapping multiple frames to a
+single NCO).
 
-.. code-block: none
+Frame Construction
+~~~~~~~~~~~~~~~~~~
 
-   const driveframe = frameof("drive", %0);
+Frames are purely virtual and can be constructed using the ``Frame`` command
 
-Second, frames return references not values. This means that calling the same
-``frameof`` command in different places is the same as assigning the frame to a
-constant and referencing it multiple times.
+.. code-block:: javascript
 
-Frame manipulation
-~~~~~~~~~~~~~~~~~~
+  frame driveframe = Frame(5e9, 0.0);
 
-The (frequency, phase, scale) of a frame can be manipulated throughout program
-by referencing ``.frequency``, ``.phase``, and ``.scale``. Operations must be 
-appropriate for the respective type, ``float`` for frequency, ``angle`` for
-phase, and ``float`` for scale. Again, the exact precision of these calculations
-is hardware specific.
+When instantiated, the frame time starts at 0. ``Frame``s can also be copied using the
+``copy`` command with argument replacement
 
-Here's an example of manipulating the phase to calibrate an ``rz`` gate:
+.. code-block:: javascript
 
-.. code-block:: none
+  frame driveframe1 = Frame(1.0, 5e9, 0.0);
```

Comment:
Why capital? To differentiate the type and initializer?

Reply:
Yes; other suggestions welcome.